sql/stats: TestDefaultColumns failed

sql/stats.TestDefaultColumns [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5205879&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5205879&tab=artifacts#/) on master @ [ce66acdbba801de88f1dd645eaedeb3834f23dbd](https://github.com/cockroachdb/cockroach/commits/ce66acdbba801de88f1dd645eaedeb3834f23dbd):


```
=== RUN   TestDefaultColumns
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/68790eb7cfaf4e8cc805e2d551cc66f6/logTestDefaultColumns4214364350
    test_log_scope.go:80: use -show-logs to present logs inline
    automatic_stats_test.go:696: query 'SELECT count(*) FROM [SHOW STATISTICS FOR TABLE t.a] WHERE statistics_name = 's'': expected:
        101
        
        got:
        41
        
    automatic_stats_test.go:701: -- test log scope end --
--- FAIL: TestDefaultColumns (66.31s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDefaultColumns.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/stats.TestDefaultColumns [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5228829?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5228829?buildTab=artifacts#/) on master @ [7dd1c51f6b5802e32bafd82e46747f349836592f](https://github.com/cockroachdb/cockroach/commits/7dd1c51f6b5802e32bafd82e46747f349836592f):


```
=== RUN   TestDefaultColumns
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/68790eb7cfaf4e8cc805e2d551cc66f6/logTestDefaultColumns1681085004
    test_log_scope.go:80: use -show-logs to present logs inline
    automatic_stats_test.go:696: query 'SELECT count(*) FROM [SHOW STATISTICS FOR TABLE t.a] WHERE statistics_name = 's'': expected:
        101
        
        got:
        44
        
    automatic_stats_test.go:701: -- test log scope end --
--- FAIL: TestDefaultColumns (62.75s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDefaultColumns.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/stats.TestDefaultColumns [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5255766?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5255766?buildTab=artifacts#/) on master @ [48e48db89eb3cd6bc38f3631364c516181811280](https://github.com/cockroachdb/cockroach/commits/48e48db89eb3cd6bc38f3631364c516181811280):


```
=== RUN   TestDefaultColumns
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/68790eb7cfaf4e8cc805e2d551cc66f6/logTestDefaultColumns415704010
    test_log_scope.go:80: use -show-logs to present logs inline
    automatic_stats_test.go:696: query 'SELECT count(*) FROM [SHOW STATISTICS FOR TABLE t.a] WHERE statistics_name = 's'': expected:
        101
        
        got:
        42
        
    automatic_stats_test.go:701: -- test log scope end --
--- FAIL: TestDefaultColumns (65.58s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDefaultColumns.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/stats.TestDefaultColumns [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5303449?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5303449?buildTab=artifacts#/) on master @ [1e2cc61b58dc14386bb68dca59814874648931c2](https://github.com/cockroachdb/cockroach/commits/1e2cc61b58dc14386bb68dca59814874648931c2):


```
        __auto__, {c18}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996169461761
        __auto__, {c19}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996214714369
        __auto__, {c20}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996262653953
        __auto__, {c21}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996306628609
        __auto__, {c22}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996383567873
        __auto__, {c23}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996456640513
        __auto__, {c24}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996505104385
        __auto__, {c25}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996549177345
        __auto__, {c26}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996593446913
        __auto__, {c27}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996636438529
        __auto__, {c28}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996679757825
        __auto__, {c29}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996724977665
        __auto__, {c30}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996770426881
        __auto__, {c31}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996816007169
        __auto__, {c32}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996855230465
        __auto__, {c33}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996899958785
        __auto__, {c34}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996945833985
        __auto__, {c35}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645996989579265
        __auto__, {c36}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997050822657
        __auto__, {c37}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997095747585
        __auto__, {c38}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997140606977
        __auto__, {c39}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997193265153
        __auto__, {c40}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997236387841
        __auto__, {c41}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997282066433
        __auto__, {c42}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997327384577
        __auto__, {c43}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997373784065
        __auto__, {c44}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997422575617
        __auto__, {c45}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997466255361
        __auto__, {c46}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997510885377
        __auto__, {c47}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997556727809
        __auto__, {c48}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997606240257
        __auto__, {c49}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997656047617
        __auto__, {c50}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997705560065
        __auto__, {c51}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997753565185
        __auto__, {c52}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997807206401
        __auto__, {c53}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997855735809
        __auto__, {c54}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997905248257
        __auto__, {c55}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997951877121
        __auto__, {c56}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645997998931969
        __auto__, {c57}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645998047068161
        __auto__, {c58}, 2022-05-28 08:36:53.404574 +0000 +0000, 0, 0, 0, 0, 765645998088945665
        
    automatic_stats_test.go:707: query 'SELECT count(*) FROM [SHOW STATISTICS FOR TABLE t.a] WHERE statistics_name = 's'': expected:
        101
        
        got:
        42
        
    automatic_stats_test.go:712: -- test log scope end --
--- FAIL: TestDefaultColumns (65.35s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDefaultColumns.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/stats/automatic_stats_test.go
+++ b/pkg/sql/stats/automatic_stats_test.go
@@ -681,7 +681,8 @@ func TestDefaultColumns(t *testing.T) {
 	sqlRun := sqlutils.MakeSQLRunner(sqlDB)
 	sqlRun.Exec(t,
 		`CREATE DATABASE t;
-		CREATE TABLE t.a (c0 INT PRIMARY KEY);`)
+		CREATE TABLE t.a (c0 INT PRIMARY KEY)
+		WITH (sql_stats_automatic_collection_enabled = false);`)
 
 	for i := 1; i < 110; i++ {
 		// Add more columns than we will collect stats on.
@@ -690,17 +691,6 @@ func TestDefaultColumns(t *testing.T) {
 	}
 
 	sqlRun.Exec(t, `CREATE STATISTICS s FROM t.a`)
-
-	// TODO(rytaft): this extra logging was added to help debug issue #81513.
-	// Remove it once that issue is resolved.
-	// === BEGINNING OF EXTRA LOGGING ===
-	res := sqlRun.QueryStr(t, `SHOW CREATE TABLE t.a`)
-	t.Log(sqlutils.MatrixToStr(res))
-	res = sqlRun.QueryStr(t, `EXPLAIN (DISTSQL) CREATE STATISTICS s FROM t.a`)
-	t.Log(sqlutils.MatrixToStr(res))
-	res = sqlRun.QueryStr(t, `SHOW STATISTICS FOR TABLE t.a`)
-	t.Log(sqlutils.MatrixToStr(res))
-	// === END OF EXTRA LOGGING ===
 
 	// There should be 101 stats. One for the primary index, plus 100 other
 	// columns.

